.orchestra/support/revng-distributable/install.sh

```diff
@@ -0,0 +1,125 @@
+
+EXCLUDE_LIST=$(mktemp -p "${TMP_ROOT}")
+
+while IFS= read -r index_file; do
+  if [ -e "$index_file" ]; then
+    grep -vE '(((libgcc_s\.so|libc\+\+\.so|libc\+\+abi\.so)(\.\d+)*)|llvm-link|llc)' "$index_file"
+  fi
+done <<< "$index_files" | sort > "$EXCLUDE_LIST"
+
+FILE_LIST=$(mktemp -p "${TMP_ROOT}")
+
+comm -23 <(find "$ORCHESTRA_ROOT" \( -type f -or -type l \) -printf '%P\n' | sort) "$EXCLUDE_LIST" | \
+grep -Ev -e '^revng/' \
+         -e '^share/revng/qa/' \
+         -e '^lib64/libclang.*\.so' \
+         -e 'lto-dump' \
+         -e '\.mypy_cache' -e '__pycache__' | \
+grep -E -e "^bin/" \
+        -e "^include/revng/PipelineC/" \
+        -e "^lib$" \
+        -e "^lib64/python/" \
+        -e "^lib64/[^/]*\.(bc|so)$" \
+        -e "^lib64/[^/]*\.so\.[^/]*$" \
+        -e "^lib64/gcc/x86_64-pc-linux-gnu/11.2.0/[^/]*\.(o|a)$" \
+        -e "^lib64/llvm/clang-release/lib/[^/]*\.so\.[^/]*$" \
+        -e "^lib64/llvm/clang-release/bin/llc$" \
+        -e "^lib64/llvm/clang-release/bin/llvm-link$" \
+        -e "^lib64/revng/analyses/[^/]*\.so$" \
+        -e "^libexec/gcc/x86_64-pc-linux-gnu" \
+        -e "^libexec/revng/" \
+        -e "^share/orchestra/" \
+        -e "^share/revng/pipelines/" \
+        -e "^share/revng/[^/]*\.ll$" \
+        -e "^x86_64-pc-linux-gnu/" | \
+sed 's|^|/|' > "$FILE_LIST"
+
+rsync \
+  --archive \
+  --verbose \
+  --progress \
+  --files-from="$FILE_LIST" \
+  "$ORCHESTRA_ROOT/." \
+  "${TMP_ROOT}${ORCHESTRA_ROOT}/revng/root/."
+
+cd "$TMP_ROOT/$ORCHESTRA_ROOT/revng"
+
+echo "Creating environment"
+cp -a "$ORCHESTRA_DOTDIR/support/revng-distributable/environment" environment
+
+# shellcheck disable=SC2094
+orchestra environment | \
+  grep -E "^export (PATH|PKG_CONFIG_PATH|PYTHONPATH|LIBRARY_PATH)" \
+    >> environment
+
+cat >> environment <<EOF
+unset ORCHESTRA_DOTDIR
+unset ORCHESTRA_ROOT
+EOF
+
+echo "Copying README.md"
+cp -a "$ORCHESTRA_DOTDIR/support/revng-distributable/README.md" .
+
+echo "Copying revng"
+cp -a "$ORCHESTRA_DOTDIR/support/revng-distributable/revng" revng
+
+
+cd "$TMP_ROOT/$ORCHESTRA_ROOT/revng/root"
+
+echo "Copying revng-distributable scripts"
+cp -a \
+  "$ORCHESTRA_DOTDIR/support/revng-distributable/revng-update" \
+  "$ORCHESTRA_DOTDIR/support/revng-distributable/revng-system-info" \
+  bin/
+
+echo "Copying install-revng-dependencies"
+cp -a "$ORCHESTRA_DOTDIR/support/revng-distributable/dockers/install-revng-dependencies" bin/
+
+echo "revng-distributable updater info"
+mkdir -p share/revng-distributable
+echo 1 > share/revng-distributable/version
+
+
+echo "Stripping components"
+STRIP_FILE_LIST=$(mktemp -p "${TMP_ROOT}")
+find 'share/orchestra' -type f -name '*.idx' ! -name '*revng*' -exec cat {} \; > "$STRIP_FILE_LIST"
+find 'libexec/revng' -type f >> "$STRIP_FILE_LIST"
+while read -r FILE; do
+  if [ -f "$FILE" ] && file "$FILE" | grep -qE 'ELF.*x86-64.*(shared|dynamic).*not stripped'; then
+    echo strip "$FILE"
+    strip "$FILE"
+  fi
+done < "$STRIP_FILE_LIST"
```

Comment:
Do we still get nice stack traces if we do this? If not, we can't do this.

Reply:
We get line numbers + filename for our libraries
LLVM libraries just get the demangled function name